### ISSUE TITLE
[Core] New pages to explain context methods and variables available in `packages.yml` and `dependencies.yml`

### ISSUE THREAD

**Issue by dbeatty10** [fishtown-analytics/docs.getdbt.com#5863]
### Link to the page(s) on docs.getdbt.com requiring updates

New pages!

### Tell us more about this update

We have pages that describe the available rendering context for these:
- [`dbt_project.yml` context](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-project-yml-context)
- [`properties.yml` context](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-properties-yml-context)
- [`profiles.yml` context](https://docs.getdbt.com/reference/dbt-jinja-functions/profiles-yml-context)
- [`on-run-end` context](https://docs.getdbt.com/reference/dbt-jinja-functions/on-run-end-context)

But not for these:
- [`packages.yml`](https://docs.getdbt.com/docs/build/packages#use-cases) (package dependencies)
- [`dependencies.yml`](https://docs.getdbt.com/docs/build/packages#use-cases) (package and cross-project dependencies)

#### TLDR
- `dependencies.yml` does not support Jinja, so there is no rendering context for that file
- `packages.yml` probably has a similar rendering context to either [`profiles.yml`](https://docs.getdbt.com/reference/dbt-jinja-functions/profiles-yml-context) or [`dbt_project.yml` context](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-project-yml-context), but we'd need to double check what it includes (or doesn't).

### Reviewers/Stakeholders/SMEs

`{{ this }}`

### Related GitHub issues

Follow-up to https://github.com/dbt-labs/docs.getdbt.com/pull/5840

**Comment by nataliefiann** [comment]
**Research done before tackling mini plan**

The [Can I define private packages in the dependencies.yml file?](https://docs.getdbt.com/faqs/Project_ref/define-private-packages) doc covers the info for `dependencies.yml`

[YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips) states you can use `env_var()` in any dbt YAML file that supports Jinja, but `profiles.yml` and `packages.yml` treat environment variables as secrets (using the `DBT_ENV_SECRET_` prefix)

Additionally, in the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) it states `packages.yml` supports Jinja rendering for historical reasons

The expandable I'd like to update is a snippet which exists on the Packages page as well

```
import UseCaseInfo from '/snippets/_packages_or_dependencies.md';

<UseCaseInfo/>

```
[Slack thread](https://dbt-labs.slack.com/archives/C09BLTL1HHA/p1763491074584449)

**Comment by nataliefiann** [comment]
**Documentation plan**

## New and updated pages

- I will update the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) expandable

What pages will you add, and what will they contain?

- I will create a new page titled "About packages.yml context" 
- A card will be created on the [dbt Jinja functions](https://docs.getdbt.com/reference/dbt-jinja-functions) page
- I will add the following intro to the new page:
<IntroText>

The following rendering context methods are available when configuring a packages.yml file. (Note, I've tested the variables from the [About dbt_project.yml](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-project-yml-context) page and they're not supported in `packages.yml`

</IntroText>

- I will add this H4 header "Available context methods":
       - [env_var](https://docs.getdbt.com/reference/dbt-jinja-functions/env_var)
       - and make a sub bullet explaining "[YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips) states you can use `env_var()` in any dbt YAML file that supports Jinja, but `profiles.yml` and `packages.yml` treat environment variables as secrets (using the `DBT_ENV_SECRET_` prefix)"

- I have tested the target variable in my instance and this is not supported. I get the following error:
```
Encountered an error:
Compilation Error
  Could not render {{ target.name }}: 'target' is undefined
```
- I have tested the `dbt_version` variable and it can't be used in the `packages.yml` as it returns a compilation error, however users can create a macro.sql
```
-- macros/get_version.sql
{% macro get_version() %}
  {% do log("The installed version of dbt is: " ~ dbt_version, info=true) %}
{% endmacro %}

```
- I've also tested builtins and it returns a compilation error as it's not supported:
```
12:00:57
Cloud CLI invocation created: 3bc8d4ab-d127-4127-b4f9-7993afccd861
12:00:57
Running dbt...
12:00:57
Encountered an error:
Compilation Error
  Could not render {{ builtins.ref('some_model') }}: 'dict object' has no attribute 'ref'
```


### [Add or edit] Page name

- I was thinking to add this sentence underneath the final bullet:
"`packages.yml` treat environment variables as secrets (using the DBT_ENV_SECRET_ prefix)" on the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) expandable

- I think it'll be a good idea to add a code example (example below) on the new page
or, it may be a good idea to link out to the YAML tips page

## Code example
include a code example if appropriate

```

"{{ env_var('MY_ENV_VAR') }}"

```

## Testing

- N/A

## Questions

What questions do you have for a writer or a developer?

1. Since YAML tips covers the `packages.yml` info, I feel it may be best to link out to that page after the final bullet in the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) section. Do you agree?

2. Is my understanding of context variables correct? 
A context variable is information that dbt automatically provides while reading a file, so you can reference it instead of hard-coding values

3. The sentence I'd like to add is also a resuasable snippet, am I correct in thinking you can't use a snippet within a snippet?

**Comment by mirnawong1** [comment]
hey @nataliefiann , great job with testing these! the plan is looking good, i think the main focus is on the packages.yml page more so than expandable. you can then link to that page from different places (like yaml tips page and packages page, etc.) 

some things to double check:
- how did you go about testing some of this? in my testing i found that `target.name` is avail in packages.yml (see below) and so is `dbt_version` (`"{% if dbt_version >= '2.0.0' %}1.3.3{% else %}1.2.0{% endif %}"`)

```
packages:
  # Working solution for target-based package versions
  - package: dbt-labs/dbt_utils
    version: "{% if target.name == 'prod' %}1.3.3{% else %}1.2.0{% endif %}"


# temrinal output

➜  jaffle_shop git:(current) ✗ rm -rf dbt_packages package-lock.yml
➜  jaffle_shop git:(current) ✗ dbtf deps --target prod
dbt-fusion 2.0.0-preview.94
   Loading profiles.yml
  Fetching packages.yml
 Installed dbt-labs/dbt_utils: 1.3.3 # confirms 1.3.3 for prod
 Installed 1 package

============================================================ Execution Summary ============================================================
Finished 'deps' successfully for target 'prod' [988ms]
➜  jaffle_shop git:(current) ✗ rm -rf dbt_packages package-lock.yml
➜  jaffle_shop git:(current) ✗ dbtf deps --target dev
dbt-fusion 2.0.0-preview.94
   Loading profiles.yml
  Fetching packages.yml
 Installed dbt-labs/dbt_utils: 1.2.0 # confirms 1.2.0 for dev
 Installed 1 package

=========================================================== Errors and Warnings ===========================================================
warning: dbt1065: Package 'dbt_utils' requires dbt version [>=1.3.0, <2.0.0], but current version is 2.0.0-preview.94. This package may not be compatible with your dbt version.
warning: dbt1065: Updated version available for dbt_utils@1.2.0: 1.3.3

============================================================ Execution Summary ============================================================
Finished 'deps' with 2 warnings for target 'dev' [798ms]
```
- for your questions:
  - Yep that makes sense and i'd probably also link to the new packages.yml context page.
- yes your understanding of context variables are right
- yes i _think_ you can use a snippet within a snippet. have you tried testing this? where are you thinking of adding a snippet?

**Comment by nataliefiann** [comment]
Re-testing

- I retested the target context variable in my packages.yml using:

```
packages:
  - package: dbt-labs/codegen
    version: "{% if env_var('DBT_ENV_NAME') == 'prod' %}0.14.0{% else %}0.13.1{% endif %}"
```

and got a successful output

- I also retested the builtins context variable with the following two examples:

```
packages:
  - package: dbt-labs/codegen
    version: "{% if builtins is defined and builtins.env_var('DBT_ENV_NAME', 'dev') == 'prod' %}0.14.0{% else %}0.13.1{% endif %}"
```

and

```
packages:
  - package: dbt-labs/codegen
    version: "{% if builtins is defined %}0.14.0{% else %}0.13.1{% endif %}"
```

and got a successful output.

- Lastly, I retested the dbt_version context variable:

```
packages:
  - package: dbt-labs/codegen
    version: "{% if dbt_version is defined %}0.14.0{% else %}0.13.1{% endif %}"

```

and got a successful output

**Comment by mirnawong1** [comment]
hey @nataliefiann , to address your qusetion below,  i also tested in my non fusion sandbox and it was fine. something like this would work in fusion or a dbt core version since it's packages and jinja functions. i see you've tested it and it worked - which is good to hear! you will probably need to see which context methods and context variables are supported in packages.yml?

<img width="1384" height="80" alt="Image" src="https://github.com/user-attachments/assets/a699c27a-4e24-4078-96b0-8680f5a19df3" />

**Comment by mirnawong1** [comment]
or add the ones you know work, and then test and add to the docs as you go?

**Comment by nataliefiann** [comment]
**Documentation plan**

## New and updated pages
What pages will you update and how will you edit them?

- **Rolling in Mirna's feedback**: I will update the [Packages](https://docs.getdbt.com/docs/build/packages#configuration) page.
- I will update the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) expandable. I will link out to the [YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips) page


What pages will you add, and what will they contain?

- **Rolling in Mirna's feedback**: I will create a new page titled "About packages.yml"

### [Add or edit] Page name

On the new page, I will have the following info:

The following rendering context methods and variables are available when configuring a `packages.yml `file. (Note, I've retested the variables from the [About dbt_project.yml](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-project-yml-context) page and they're supported in `packages.yml`

(H4 header) "Available context methods":

- [env_var](https://docs.getdbt.com/reference/dbt-jinja-functions/env_var)
- and make a sub bullet explaining "[YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips) states you can use env_var() in any dbt YAML file that supports Jinja, but profiles.yml and packages.yml treat environment variables as secrets (using the DBT_ENV_SECRET_ prefix)" **This is the snippet within a snippet I was referring to. I haven't tested this but I'll test this in VSCode**

(H4 header) "Available context variables":
- [target](https://docs.getdbt.com/reference/dbt-jinja-functions/target)
- [builtins](https://docs.getdbt.com/reference/dbt-jinja-functions/builtins)
- [dbt_version](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt_version)

- I will include a related docs section and link out to:
        - [When to use package dependencies](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml)
        - [YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips)
        - [Packages](https://docs.getdbt.com/docs/build/packages)

## Code example

- I will see how the example below looks on the page

`"{{ env_var('MY_ENV_VAR') }}"
`

## Testing

- above

## Questions

What questions do you have for a writer or a developer?

1. Is my new plan ok?
2. I'm a bit stuck on where the new page should live - I was thinking a possible place could be somewhere within [Build dbt projects](https://docs.getdbt.com/docs/build/projects) but would appreciate your thoughts

**Comment by mirnawong1** [comment]
hey @nataliefiann thanks for this and looks good! 

> I will update the [Packages](https://docs.getdbt.com/docs/build/packages#configuration) page.
 - what update will you be making to the [packages doc](https://docs.getdbt.com/docs/build/packages#configuration) ? sorry i might've missed it if you mentioned it earlier
- would love to hear your thoughts on why you're thinking of the 'build dbt projects' page? can you explain a bit more so i can share thoughts.

**Comment by mirnawong1** [comment]
> What pages will you add, and what will they contain?
> 
> * I will create a new page titled "About packages.yml context"
> * A card will be created on the [dbt Jinja functions](https://docs.getdbt.com/reference/dbt-jinja-functions) page
> * I will add the following intro to the new page:

hey @nataliefiann check out your earlier plan where you talk about where it should live - i think that's the right place since it addresses DOug's original feedback. but would love to hear why you're thinking build projects

**Comment by nataliefiann** [comment]
Hiya @mirnawong1 

Thanks for reviewing this again for me.

Apologies - I made the bullet underneath "I will update the [Packages (https://docs.getdbt.com/docs/build/packages#configuration) page" to indicate the change I'd be making. I wanted to update the [When to use package dependencies?](https://docs.getdbt.com/docs/mesh/govern/project-dependencies#about-packagesyml-and-dependenciesyml) expandable by linking it out to the [YAML tips](https://docs.getdbt.com/docs/build/dbt-tips#yaml-tips) page. 
Just to note, that expandable is a snippet so I will need to update the snippet.

I was a bit unsure if the new doc should live in the [dbt Jinja functions and context variables](https://docs.getdbt.com/reference/dbt-jinja-functions-context-variables) page as (from my understanding, I could be wrong), packages.yml isn't a jinja function or context variable. I thought it might cause some reader confusion 

I thought [Build dbt projects](https://docs.getdbt.com/docs/build/projects) might be a more suitable location as they're used to define package dependencies for a project.

**Comment by mirnawong1** [comment]
hey @nataliefiann , ah ok. the reason i'm not sure it shoudl live under 'build dbt projects' is because yes even though packages.yml isn’t a jinja function, it is jinja-rendered, which matches the scope of the issue and the existing context variable pages. 

i'm interpreting this as 'the user need here is not “where do I define package dependencies?". it’s: “What Jinja context is available (if any) when dbt renders packages.yml?”

a separate question to this i have is do you think we should add a short note in the jinja functions/context variables landing page somewhere stating `dependencies.yml: Jinja not supported`

**Comment by mirnawong1** [comment]
but if you feel strongly it should be in build dbt projeccts, that's totally fine and you can add it to the pr. we can always get a second opinion as a gut check 🙏

**Comment by nataliefiann** [comment]
Thanks @mirnawong1 

I'm happy to go with your recommendation, the fact packages.yml is jinja rendered makes sense for the page to live on [dbt Jinja functions and context variables](https://docs.getdbt.com/reference/dbt-jinja-functions-context-variables) page

Yes, I agree adding a note on the page to state dependencies.yml isn't Jinja supported. I will test a few things as I'm writing up the PR
